src/shogun/mathematics/linalg/BaseVector.h

```diff
@@ -54,7 +54,7 @@ struct BaseVector
 	}
 
 	/** Data Storage */
-	virtual bool onGPU() = 0 ;
+	virtual bool onGPU() const = 0 ;
```

Comment:
forgive my ignorance but if we have `CPUVector` and `GPUVector` then why do we need `onGPU` function. since the class could identify the storage or?